### PR TITLE
Update TYPED_TEST_CASE to TYPED_TEST_SUITE with fallback for older GTest versions

### LINKED ISSUE #3373: GTest deprecated `TYPED_TEST_CASE`
Warning is as follows:
```
pcl/test/common/test_geometry.cpp:49:1: warning: 'TypedTestCaseIsDeprecated' is deprecated: TYPED_TEST_CASE is deprecated, please use TYPED_TEST_SUITE [-Wdeprecated-declarations]
TYPED_TEST_CASE(XYZPointTypesTest, XYZPointTypes);
^
```
This occurs in many places throughout tests.

## Your Environment
* Operating System and version: Linux
* Compiler: clang 8.0.1
* PCL Version: master

The GTest used is from the [master branch](https://github.com/google/googletest) as per the "gtest" best-practices instead of OS installed version.

## Expected Behavior
No warnings

## Possible Solution
A simple `sed` might solve this

### PR DESCRIPTION
**Pull request by shrijitsingh99**
An updated PR which builds upon #3419 and fixes the remaining issues
Fixes #3373 , closes #3419
Credits to @jammm, all of the work was already done by him, I just had to fix some minor issues

### CODE REVIEW

File: test/include/pcl/test/test_macros.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#endif
+
+/**
+ * \brief Macro choose between INSTANTIATE_TEST_CASE_P and INSTANTIATE_TEST_SUITE_P depending on the GTest version
+ *
+ * \ingroup test
+ */
+#if !defined(INSTANTIATE_TEST_SUITE_P)
+  #define INSTANTIATE_TEST_SUITE_P INSTANTIATE_TEST_CASE_P
+#endif
```
Comment (kunaltyagi) [inline test/include/pcl/test/test_macros.h]: Newline at the end of file please. That makes it a text file for Unix systems

Comment (shrijitsingh99) [reply test/include/pcl/test/test_macros.h]: Yeah sorry about that, somehow missed it.
Fixed now.

File: test/include/pcl/test/test_macros.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+#pragma once
+
+/**
+ * \file pcl/test/test_macros.h
+ *
+ * \brief Defines all the PCL test macros used
+ * \ingroup test
+ */
+
```
Comment (kunaltyagi) [inline test/include/pcl/test/test_macros.h]: I feel we should be including gtest header in this file to prevent header order dependencies downstream

Comment (kunaltyagi) [inline test/include/pcl/test/test_macros.h]: I feel we should be including gtest header in this file to prevent header order dependencies downstream

Comment (shrijitsingh99) [reply test/include/pcl/test/test_macros.h]: Yeah, that's a good idea, I will avoid any future errors and causes no harm in adding it.

**Comment by taketwo** [comment]
Sorry, me again with header naming suggestions :man_shrugging: 

As was agreed in the other discussion (#3654), umbrella-named headers like "xxx_macros" are not desirable. In the case at hand, I believe a more suitable name would be "pcl/test/gtest.h". Indeed, the file is nothing else but include of GTest plus a compatibility layer to support older versions. The fact that this compatibility layer is implemented through macros is not important.

**Comment by SergioRAgostinho** [comment]
@shrijitsingh99  Incorporate @taketwo's comment regarding file renaming and I'll review it after.

**Comment by shrijitsingh99** [comment]
> @shrijitsingh99 Incorporate @taketwo's comment regarding file renaming and I'll review it after.

Renamed the file as @taketwo suggested

**Comment by taketwo** [comment]
From maintainer perspective yes, please squash. Maybe add contributors' names to the commit message.

> What's the current established label for me waiting for other maintainers to chime?

We abolished it. Is there a need? What's the difference to "needs: review"? We can introduce "needs: maintainers feedback".

**Comment by SergioRAgostinho** [comment]
In my mind needing a  review means full code review, feedback is to just chime in on a specific topic. While navigating through the issue list it helps me distil both. Did the meaning for needs: review also change?

**Comment by taketwo** [comment]
>  Did the meaning for needs: review also change?

No, it's still the same.

> feedback is to just chime in on a specific topic.

OK, understood.

I just wanted to compress the label set, which implies a bit of meaning overloading... In this case the difference is too large though. Let's have "needs: maintainers feedback/opinions"?

**Comment by SunBlack** [comment]
Just a hint: `#include <gtest/gtest.h>` is not required anymore as it is always includes via `#include <pcl/test/gtest.h>` now

**Comment by kunaltyagi** [comment]
> Let's have "needs: maintainers feedback/opinions"?

Added the label.

> #include <gtest/gtest.h> is not required anymore

Ah yeah, I guess I'll create an issue of that

**Pull request merged**